src/cincinnati/client.rs

```diff
@@ -34,6 +34,33 @@ pub struct Graph {
     pub edges: Vec<(u64, u64)>,
 }
 
+/// Error related to the graph service.
+#[derive(Clone, Debug, Fail, Deserialize, Serialize, PartialEq, Eq)]
+pub struct GraphError {
+    /// HTTP status code returned by the server.
+    #[serde(skip)]
+    pub(crate) status: Option<reqwest::StatusCode>,
+    /// Machine-friendly brief error kind.
+    pub(crate) kind: String,
+    /// Human-friendly detailed error explanation.
+    pub(crate) value: String,
```

Comment:
Optional: hmm, sounds like we could group them into an enum? E.g. each member being a different error and have `Display` be the human-friendly one and an impl function for the machine one.

Reply:
I'm not sure I'm following you on this one. Mind writing down the enum variants you are thinking about? Also, do note that in some cases `kind` is dynamic (e.g. the ones derived from a raw status code).

Reply:
Something like:

```rust
enum GraphError {
    HTTP(u32),
    JsonDecode(String),
    ...
}

impl GraphError {
    fn machine_error(&self) -> String { ... }
}

impl std::fmt::Display for GraphError {
    ...
}
```

?

Feel free to disregard if it's not worth the overhead. :)